pkg/kapp/clusterapply/waiting_changes.go

```diff
@@ -126,3 +139,12 @@ func (c *WaitingChanges) Complete() error {
 func (c *WaitingChanges) stats() string {
 	return fmt.Sprintf("[%d/%d done]", c.numWaited, c.numTotal)
 }
+
+func checkForTimeout(change WaitingChange, timeout time.Duration, err error) error {
+	if time.Now().Sub(change.startTime) > timeout {
+		return errors.New("resource wait timeout")
+	} else if err != nil {
+		return err
+	}
```

Comment:
we should prefer "actual" error over timeout error when we have it.

Reply:
Removed this function and making sure actual error will get the preference always over timeout